### PR TITLE
conversations: sort thread by createdmodseq if same internaldate

### PR DESCRIPTION
**Pull request by brong**
This gives us a stable sort when multiple external messages are appended very quickly, giving them the same internaldate.

This is being added without a new version number, as it extends a data structure in a backwards and forwards compatible way.  If not set, the CREATEDMODSEQ field will be zero, which is fine.  This will still lead to correct sorting for newly added emails on any thread.  The only slight weirdness might be if an existing message in a thread gets moved or copied into a new folder, which will lead to it getting a createdmodseq value, and hence sorting to the end for that internaldate. This will be very rare, and a convdb rebuild will fix it.

### CODE REVIEW

**Review by rsto — COMMENTED** [review]
This looks good. Can you add a test for that?

**Comment by rsto** [comment]
I will add createdmodseq support for Email/get on top of this PR. I will then add a test for Thread/get, too.

**Comment by rsto** [comment]
Now with support for createdModseq in Email/get. Removing myself as reviewer, as I wrote the Email/get part of this. I did not come up with a good test for Thread/get so this will need to do.

File: cassandane/tiny-tests/JMAPEmail/email_get_createdmodseq [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+test
+EOF
+    $mimeMessage =~ s/\r?\n/\r\n/gs;
+
+    # Sleep until the start of a second. This should result
+    # in the duplicates having the same internal date,
+    # unless the test code is running really slow.
+    my ($now_secs, $now_microsecs) = gettimeofday;
+    usleep(2 * 1000000 - $now_microsecs);
```
Comment (ksmurchison) [inline cassandane/tiny-tests/JMAPEmail/email_get_createdmodseq:41]: Why not just set the internaldate as part of the APPEND?

**Review by ksmurchison — COMMENTED** [review]
LGTM other than my comment.  Perhaps IMAPTalk doesn't support setting internaldate in APPEND?

File: cassandane/tiny-tests/JMAPEmail/email_get_createdmodseq (hunk 1 above)
Comment (rsto) [reply cassandane/tiny-tests/JMAPEmail/email_get_createdmodseq]: Because I was not aware that this is an option. I updated the test code.

**Review by ksmurchison — APPROVED** [review]

**Comment by brong** [comment]
> LGTM other than my comment. Perhaps IMAPTalk doesn't support setting internaldate in APPEND?

I'm pretty sure it does!

```
my @res = $Talk->append($name, $flagtext, { Quote => $InternalDate }, $fh);
my @res = $Talk->append($name, $flagtext, { Quote => $InternalDate }, { Literal => $content });
```

**Comment by rsto** [comment]
Yep, the test got updated to use that in APPEND.

**Pull request merged**